src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -22,13 +22,85 @@ public sealed class TelemetryConfiguration : IDisposable
         private static TelemetryConfiguration active;
 
         private readonly SnapshottingList<ITelemetryInitializer> telemetryInitializers = new SnapshottingList<ITelemetryInitializer>();
+        private ITelemetryChannel telemetryChannel = null;
         private TelemetryProcessorChain telemetryProcessorChain;
         private string instrumentationKey = string.Empty;
         private bool disableTelemetry = false;
         private TelemetryProcessorChainBuilder builder;
         private SnapshottingList<IMetricProcessor> metricProcessors = new SnapshottingList<IMetricProcessor>();
 
         /// <summary>
+        /// Indicates if this instance has been disposed of.
+        /// </summary>
+        private bool isDisposed = false;
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        public TelemetryConfiguration() : this(new InMemoryChannel())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        /// <param name="instrumentationKey">The instrumentation key this configuration instance will provide.</param>
+        public TelemetryConfiguration(string instrumentationKey) : this(new InMemoryChannel())
+        {
+            if (instrumentationKey == null)
+            {
+                throw new ArgumentNullException("instrumentationKey");
+            }
+
+            this.instrumentationKey = instrumentationKey;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the TelemetryConfiguration class.
+        /// </summary>
+        /// <param name="channel">The telemetry channel to provide with this configuration instance.</param>
+        public TelemetryConfiguration(ITelemetryChannel channel)
```

Comment:
Thanks for the explanation - it helps. I don't know if it still holds true, but a while back, a missing `InstrumentationKey` was a common source of missing telemetry. As a user of this API, I'd say that indicating that the instrumentation key is optional is misleading. Unless you have strong customer evidence that says otherwise, I'd suggest dropping the `TelemetryConfiguration(ITelemetryChannel)` constructor for now.